### PR TITLE
Hackathon refactor the CellSetTool tests to use the real Redux store

### PR DESCRIPTION
**Pull request by ivababukova**
# Background
Inspired by:
https://medium.com/@lucksp_22012/dont-use-mock-store-use-your-real-store-b319d7e4e22e

#### Link to staging deployment URL 
https://ui-redux-tests.scp-staging.biomage.net/

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes
1. Moved things that we mock for each component into a global `setupTests` mock. Lots of the file changes are due to this.
2. Rewrote the CellSetTool test using the real Redux store.
3. Found a very small bug in the cellSetOperations and deleted some unneeded code in the code that generates the cell set hierarchy.

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [x] Unit tests written
- [x] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [x] Deployed to staging
- [ ] Passed integration tests 

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: src/utils/cellSetOperations.js [hunk 1]
```diff
@@ -25,6 +25,10 @@ const intersection = (listOfSets, properties) => {
     (set) => set && set.size > 0,
   );
 
+  if (sets.length === 0) {
+    return new Set();
+  }
+
```
Comment (ivababukova) [inline src/utils/cellSetOperations.js:31]: I actually found this bug while testing

File: src/redux/actions/cellSets/loadCellSets.js [hunk 2]
```diff
@@ -9,18 +9,16 @@ const loadCellSets = (experimentId, forceReload = false) => async (dispatch, get
...
     return null;
   }
-
   if (getState().cellSets.error) {
     dispatch({
       type: CELL_SETS_LOADING,
     });
   }
 
-  const sampleIds = { sample: getState().experimentSettings.info.sampleIds };
```
Comment (ivababukova) [inline src/redux/actions/cellSets/loadCellSets.js:22]: Check if `info.sampleIds` is used anywhere in the code, if not we should remove it.

File: src/utils/environment.js [hunk 3]
```diff
@@ -9,6 +9,11 @@ const Environment = {
 const ssrGetCurrentEnvironment = () => {
   let currentEnvironment = null;
 
+  console.log('*******k8s  ', process.env.K8S_ENV);
```
Comment (ivababukova) [inline src/utils/environment.js]: remove this print

File: src/redux/actions/cellSets/loadCellSets.js (hunk 2 above)
Comment (ivababukova) [reply src/redux/actions/cellSets/loadCellSets.js:22]: I will make a small ticket for this and we can do it after I merge these changes in.

Comment (ivababukova) [reply src/redux/actions/cellSets/loadCellSets.js:22]: I spoke with Agi, we still need it: it is used in Data-Processing index.jsx page. I won't be removing it.

File: src/__test__/components/data-exploration/cell-sets-tool/CellSetsTool.test.jsx [hunk 4]
```diff
@@ -181,330 +122,398 @@ describe('CellSetsTool', () => {
...
-      {
-        ...storeState,
-        cellSets: {
-          ...storeState.cellSets,
-          selected: { ...storeState.cellSets.selected, cellSets: ['cluster-a', 'cluster-b', 'cluster-c'] },
-        },
-      },
-    );
+    // test the union cellSet function
+    const hardcodedUnion = new Set([12, 13, 14, 15]);
```
Comment (cosa65) [inline src/__test__/components/data-exploration/cell-sets-tool/CellSetsTool.test.jsx]: maybe call it `expectedUnion`?

**Review by cosa65 — APPROVED** [review]

**Pull request merged**